Objective-C/Tests/ReplicatorTest+CustomConflict.m

```diff
@@ -233,6 +241,74 @@ - (void) testConflictResolverMergeDoc {
+    NSMutableArray* conflictedDocs = [NSMutableArray array];
+    
+    // higher remote revID
+    NSString* docID = @"doc1";
+    [self makeConflictFor: docID withLocal: localData withRemote: remoteData localFirst: YES];
+    [conflictedDocs addObject: @[[self.db documentWithID: docID],
+                                 [otherDB documentWithID: docID]]];
+    
+    // higher local revID
+    docID = @"doc2";
+    [self makeConflictFor: docID withLocal: localData withRemote: remoteData localFirst: NO];
+    [conflictedDocs addObject: @[[self.db documentWithID: docID],
+                                 [otherDB documentWithID: docID]]];
+    
+    // higher generation
+    docID = @"doc3";
+    [self makeConflictFor: docID withLocal: localData withRemote: remoteData localFirst: YES];
+    CBLMutableDocument* doc = [[self.db documentWithID: docID] toMutable];
+    [doc setValue: @"value3" forKey: @"key3"];
+    [self saveDocument: doc];
+    [conflictedDocs addObject: @[[self.db documentWithID: docID],
+                                 [otherDB documentWithID: docID]]];
+    
+    // delete local
+    docID = @"doc3";
+    [self makeConflictFor: docID withLocal: localData withRemote: remoteData localFirst: YES];
+    [self.db deleteDocument: [self.db documentWithID: docID] error: &error];
+    [conflictedDocs addObject: @[[self.db documentWithID: docID] ?: [NSNull null],
+                                 [otherDB documentWithID: docID]]];
+    
+    // delete remote
+    docID = @"doc4";
+    [self makeConflictFor: docID withLocal: localData withRemote: remoteData localFirst: YES];
+    [otherDB deleteDocument: [otherDB documentWithID: docID] error: &error];
+    [conflictedDocs addObject: @[[self.db documentWithID: docID],
+                                 [otherDB documentWithID: docID] ?: [NSNull null]]];
+    
+    
+    CBLReplicatorConfiguration* pullConfig = [self pullConfig];
+    [self run: pullConfig errorCode: 0 errorDomain: nil];
+    
+    for (NSArray* docs in conflictedDocs) {
+        CBLDocument* doc1 = docs[0];
+        CBLDocument* doc2 = docs[1];
+        
+        // if any deleted, success revision should be deleted.
+        if ([doc1 isEqual: [NSNull null]] || [doc2 isEqual: [NSNull null]]) {
+            docID = [doc1 isKindOfClass: [NSNull class]] ? doc2.id : doc1.id;
+            CBLDocument* successDoc = [[CBLDocument alloc] initWithDatabase: self.db
+                                                                 documentID: docID
+                                                             includeDeleted: YES
+                                                                      error: &error];
+            AssertNil(error);
+            Assert(successDoc.isDeleted);
+            continue;
+        }
+        
+        if (doc1.generation != doc2.generation)
+            Assert(doc1.generation > doc2.generation);
+        else
+            Assert([doc1.revID compare: doc2.revID]);
```

Comment:
This is just checking that their revIDs aren't equal; is that what you meant? If so, calling `-isEqualToString:` would be clearer. Or did you mean to test whether doc1.revID is ≥ doc2.revID?

Reply:
doc1.revID is > doc2.revID

Reply:
since this is arbitrary, this is useless comparison.